### DISCUSSION TITLE
Login error when using TLS

### DISCUSSION THREAD

**Discussion by carsoni** [minio/minio#12883] [Q&A]
**Description**
I have successfully installed and started a minio instance using the docker image and am able to access it via the console when running under HTTP

Unfortunately I'm experiencing a strange error when attempting to login through the console after I enable TLS on the minio instance.

The docker command and the mc config are listed below.

When I navigate to https://observe01.paradigm.local/login I am presented with the login page but when I attempt a login I get the following in the docker logs for the container

`    original error -> (user_login.go:50): invalid login`
`    debugging error: expected element type <AssumeRoleResponse> but have <html>`

I have tried logging in with the root user and password and with a user with a consoleAdmin policy attached with no success. I also get the same error even if I start the container with no root user or root password. The login page appears but minioadmin:minioadmin fails with the same error.

This behavior is the same whether I attempt the login from a remote machine or the one local to the container 

Note that I have no problem accessing the API using mc. I can read and list users, buckets files etc and change user policies.

I have tried all combinations of port assignments to no avail and was wondering if anyone has experienced this issue

Thanks & regards
Ian Carson

**Setup**
Minio is running in a docker container

The container is started with the following command

docker run --rm -p 9000:9000 -p 443:443 \
-e "MINIO_ROOT_USER=xxxxxxxxx" \
-e "MINIO_ROOT_PASSWORD=xxxxxxxxx" \
-e "MINIO_SERVER_URL=https://observe01.paradigm.local" \
--name miniostore 
-v /mnt/singlestore:/data 
-v /mnt/minioconfig:/root/.minio 
minio/minio server --console-address ":443" /data

The appropriate crt, key and CA files are in the proper structure in /mnt/minioconfig

When this container runs I am able to access it using the mc with this config

"miniostore": {
             "url": "https://observe01.paradigm.local:9000",
              "accessKey": "xxxxxxxxxxxxx",
              "secretKey": "xxxxxxxxxxxxxx",
               "api": "s3v4",
               "path": "auto"
}

*Note the accessKey and secretKey are the ones used in the docker run statement above*

**Comment by harshavardhana** [comment]
Make sure to set MINIO_BROWSER_REDIRECT_URL

**Reply by carsoni** [reply]
Hi @harshavardhana 

I tried adding the following 2 values as MINIO_BROWSER_REDIRECT_URL as you suggested

a) `-e "MINIO_BROWSER_REDIRECT_URL=https://observe01.paradigm.local"`
or 
b) `-e "MINIO_BROWSER_REDIRECT_URL=https://console.observe01.paradigm.local"`

but the same error persists.

If I change the port assignment from 443:443 to 9001:9001 and the --console-address to 9001 I get the error 

`Post "https://observe01.paradigm.local/" : dial tcp 192.168.0.29:443: connect : connection refused`

which I think makes sense

If I use the redirect url to point at the docker network address of the container I still get the "expected element type..." error

Your thoughts are appreciated

**Reply by carsoni** [reply]
Perhaps the debugging error gives a hint as to the problem?

`debugging error: expected element type <AssumeRoleResponse> but have <html>`

**Reply by carsoni** [reply]
In order to bypass any possible issues with browsers I have tried direct api access using Postman

`POST https://observe01.paradigm.local/api/v1/login`

using the root_user and root_password as accessKey and secretKey. 

This approach also produces the "expected element type..." error

**Reply by carsoni** [reply]
Hi @harshavardhana

I'm pretty sure this has something to do with the interaction between the redirection and the docker bridge/proxying.

Have you come across anything like that causing issues?

Regards
Ian

**Comment by carsoni** [comment]
Hi @harshavardhana 

I continue to attempt a solution to this problem. 

To that end I thought it best to remove docker from the equation and have installed Minio directly on the server (Ubuntu 20.04 LTS running in a VM on TrueNAS). With TLS certificates in the appropriate structure under ${HOME}/.minio.

Once again the server is available via mc but gives the "expected element type <AssumeRoleResponse> but have <html>" error when attempting to login via the console

I am using the browser (Firefox) on the minio server to login to minio so there is no proxy or LB involved and that should also eliminate any CORS issues.

I have tried with and without setting a MINIO_BROWSER_REDIRECT_URL using both :443 and :9005 (this last chosen at random)

The response header from the failed login attempt looks like this

`HTTP/2 401 Unauthorized
content-type: application/json
server: MinIO Console
x-content-type-options: nosniff
x-frame-options: DENY
x-xss-protection: 1; mode=block
content-length: 73
date: Fri, 06 Aug 2021 01:52:12 GMT
X-Firefox-Spdy: h2`

My TLS certificates do not specify an IP SAN so I am using MINIO_SERVER_URL 

**Example 1**
Setting console-address to :9005

If I browse to https://localhost:9005 the server landing page shows the login frame which I can fill in and press login but I get

`Post "https://observe01.paradigm.local/" : dial tcp 192.168.0.29:443: connect : connection refused`

**Example 2**
The last error message leads me to change the console-address to :443

Now I browse to https://localhost gets me to the login frame which I can fill in and press login but I get

`original error -> (user_login.go:50: invalid Login)`
`debugging error: expected element type <AssumeRoleResponse> but have <html>`

I have tried to find user_login.go to see if I could see an issue but a search for "user_login" in the minio repo failed

Given that my installation is directly on the server I don't believe there is anything between the web page and the server that could be passing html to minio other than a minio process/method. Is it possible that there is a bug in the UI especially since https access to the server is fine when using mc? 

Many thanks in advance for your help 
Regards
Ian Carson
